_source/_posts/2025-07-28-express-oauth-pkce.md

```diff
@@ -0,0 +1,700 @@
+      item: "Event Booth",
+      amount: 950,
+    },
+  ],
+  hr: [
+    {
+      name: "Eve Martinez",
+      item: "Internet",
+      amount: 300,
+    },
+    {
+      name: "Frank Wilson",
+      item: "Compliance Training",
+      amount: 600,
+    },
+    {
+      name: "Grace Li",
+      item: "Conference Travel",
+      amount: 1500,
+    },
+    {
+      name: "Henry Zhang",
+      item: "Team Offsite",
+      amount: 1000,
+    },
+  ],
+  marketing: [
+    {
+      name: "Alice Johnson",
+      item: "Payroll Processing",
+      amount: 750,
+    },
+    {
+      name: "Carol Lee",
+      item: "Compliance Training",
+      amount: 400,
+    },
+    {
+      name: "Eve Martinez",
+      item: "Team Lunch",
+      amount: 200,
+    },
+    {
+      name: "Frank Wilson",
+      item: "Team Offsite",
+      amount: 850,
+    },
+  ],
+  legal: [
+    {
+      name: "Grace Li",
+      item: "Event Booth",
+      amount: 1100,
+    },
+    {
+      name: "David Kim",
+      item: "Product Launch Campaign",
+      amount: 1300,
+    },
+    {
+      name: "Bob Smith",
+      item: "Conference Travel",
+      amount: 1250,
+    },
+    {
+      name: "Henry Zhang",
+      item: "Team Lunch",
+      amount: 170,
+    },
+  ],
+  "dev-advocacy": [
+    {
+      name: "Eve Martinez",
+      item: "Internet",
+      amount: 280,
+    },
+    {
+      name: "Frank Wilson",
+      item: "Payroll Processing",
+      amount: 720,
+    },
+    {
+      name: "Grace Li",
+      item: "Compliance Training",
+      amount: 500,
+    },
+    {
+      name: "Alice Johnson",
+      item: "Team Offsite",
+      amount: 950,
+    },
+  ],
+};
+
+export function getModifiedTeam(team) {
+  if (!team?.trim()) return [];
+
+  const toPascalCase = (str) =>
+    str
+      .trim()
+      .split(/\s+/)
+      .map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
+      .join(' ');
+
+  const toKebabCase = (str) => str.trim().toLowerCase().split(' ').join('-');
+
+  if (team === 'admin') {
+    return ALL_TEAMS_NAME.map((element) => ({
+      id: toKebabCase(element),
+      label: toPascalCase(element),
+    }));
+  }
+
+  return [
+    {
+      id: toKebabCase(team),
+      label: toPascalCase(team),
+    },
+  ];
+}
+```
+
+The file also defines `getModifiedTeam`, a helper that converts a team name into an array of objects. Each object has an id and a label. If the team is admin, the function returns an object for every entry in `ALL_TEAMS_NAME`; otherwise, it returns a single object for the specified team. Later in the project, the app calls this function to transform the user's team information.
+
+### Create a file to handle authentication
+
+Create an `auth.js` file for this step. This file uses the [openid-client](https://www.passportjs.org/packages/openid-client) library to handle the OIDC flow: it logs users in, exchanges the authorization code for tokens, and logs them out. It also defines a middleware that guards protected routes.
```

Comment:
```suggestion
Create an `auth.js` file for this step. This file uses the [openid-client](https://www.npmjs.com/package/openid-client) library to handle the OIDC flow: it logs users in, exchanges the authorization code for tokens, and logs them out. It also defines a middleware that guards protected routes.
```

Recommend either the npm or GH URL instead of the package through Passport (as we're not using it as a Passport strategy)